File: step13_eventbridge/eventbridge_with_lambda/lambda/producer.js
const AWS = require("aws-sdk");

function helper(body) {
  const eventBridge = new AWS.EventBridge({ region: "eu-west-1" });

  return eventBridge
    .putEvents({
      Entries: [
        {
          EventBusName: "default",
          Source: "custom.api",
          DetailType: "order",
          Detail: `{ "country": "${body.country}" }`,
        },
      ],
    })
    .promise();
}

exports.handler = async function (event, context) {
  console.log("EVENT BODY: \n", event.body);
  const e = await helper(JSON.parse(event.body));
  return {
    statusCode: 200,
    headers: { "Content-Type": "text/html" },
    body: `<h1>Event Published to Eventbridge</h1>${JSON.stringify(
      e,
      null,
      2
    )}`,
  };
};

File: step15_eventbridge/Python/eventbridge_with_lambda/cdk.out/asset.32d3b04c9f62130347ed0c1c56390899ecb62f29ba4fe13306606e7977206146/producer.py
// const AWS = require("aws-sdk");
import json
import boto3

# Create CloudWatchEvents client
cloudwatch_events = boto3.client('events')


def handler(event, context):
    print(event["body"])
    body = event["body"]
    print(body)
    console.log("EVENT BODY: \n", event["body"])
#  const eventBridge = new AWS.EventBridge({ region: "eu-west-1" });
    response = cloudwatch_events.put_events(
        Entries=[
            {
                'EventBusName': "default",
                'Detail': json.dumps({ "country": "{"body['country']"}"}),
                'DetailType': 'order',
                'Source': 'custom.api'
            }
        ]
    )
    print(response['Entries'])


  return {
    "statusCode": 200,
    "headers":  json.dumps({ "Content-Type": "text/html" }),
    "body": "Event Published to Eventbridge" + json.dumps(response)

}






